plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php

```diff
@@ -86,23 +86,25 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
             $output->writeln("\n" . Piwik::translate('CoreUpdater_ConsoleStartingDbUpgrade'));
 
-            foreach ($queries as $query) {
-                $output->write("\n" . 'Executing ' . $query . '... ');
-                Db::get()->exec($query);
-                $output->write(' done.');
+            try {
+                throw new \Exception();
+                foreach ($queries as $query) {
+                    $output->write("\n" . 'Executing ' . $query . '... ');
+                    Db::get()->exec($query);
+                    $output->write(' done.');
+                }
+
+                $output->writeln("\n" . 'Updating used database charset in config.ini.php.');
+                $config                      = Config::getInstance();
+                $config->database['charset'] = 'utf8mb4';
+            } finally {
+                if (!$keepTracking) {
+                    $output->writeln("\n" . Piwik::translate('Enabling Matomo Tracking'));
+                    $config->Tracker['record_statistics'] = '1';
+                    $config->forceSave();
```

Comment:
Should be outside the if statement to save the charset also?

Reply:
ah, you're right. changed that.